### ISSUE TITLE
Plugin Details: indicate when a plugin has a paid version

### ISSUE THREAD

**Issue by cpapazoglou** [Automattic/wp-calypso#60769]
Yoast premium has also a free version. While user might directly discover Yoast premium there will also be cases where users go straight ahead to the free version. 

How can we better indicate that there is a paid version too directly in Plugin Details page?

### UPDATE
- [ ] add "Premium version available", when searching the jetpack index for `org_slug` returns a result

<img width="890" alt="CleanShot 2022-05-17 at 17 35 59@2x" src="https://user-images.githubusercontent.com/12430020/168837521-92b1e38b-d4b2-43d4-ba45-a09c01ea94e6.png">

Epic Issue: https://github.com/Automattic/wp-calypso/issues/43332
P2: pdh6GB-sk-p2
Figma: f84fqtCG4pCnCh2Hq0But5-fi

**Comment by cpapazoglou** [comment]
The paid plugins definition contains a `org_slug` which we can use to match a free with a paid plugin.

![CleanShot 2022-02-18 at 12 41 29@2x](https://user-images.githubusercontent.com/12430020/154667439-2a001c8a-62dd-4fd4-854c-c8878a7eddc0.png)

**Comment by cpapazoglou** [comment]
@SaxonF, @vinimotaa 

This one will be useful for us also for adding Sensei (free) which has Sensei LMS (paid). 

https://woocommerce.com/search/?q=sensei&collections=product

![CleanShot 2022-02-23 at 20 01 03@2x](https://user-images.githubusercontent.com/12430020/155379273-76dcfe7f-9e97-41ce-be90-e5aa123e2cbb.png)


cc: @ianstewart, @allilevine

**Comment by cpapazoglou** [comment]
@vinimotaa are there any explorations here?

**Comment by vinimotaa** [comment]
Hey, @cpapazoglou this completely skipped from my books. The case here is for us to somehow indicate in the free plugin page that we have a paid version? From the product card, I see that we currently have that.


<img width="1110" alt="image" src="https://user-images.githubusercontent.com/97929097/159676463-a7502f13-b306-48ee-83cb-6eafb931697e.png">

**Comment by cpapazoglou** [comment]
@vinimotaa no worries! My comment regarding Sensei is a little bit distracting. The use case is the following:

- User visits https://wordpress.com/plugins/wordpress-seo
- If that plugin has a premium version, how might we indicate it?

**Comment by vinimotaa** [comment]
Hey @cpapazoglou, I thought that for time being, we could add a strong call to action leading users to the paid version.

There are 2 workstreams to improve this in the near future. The product card design and the product page itself.

Let me know if that would suffice for now.

<img width="1393" alt="image" src="https://user-images.githubusercontent.com/97929097/160149009-f7c1ae23-93a7-4c8e-ab60-1c2b2ca2c09c.png">

**Comment by cpapazoglou** [comment]
Thanks @vinimotaa for working on this. Yes, a CTA sounds good. I wonder how could we enhance it by using a notice (or anything) which allows for some explanatory text. example
<img width="596" alt="CleanShot 2022-03-28 at 10 51 10@2x" src="https://user-images.githubusercontent.com/12430020/160351763-d39cb858-16a3-4335-9c73-a6c358368f86.png">

**Comment by cpapazoglou** [comment]
Design here pdh6GB-sk-p2

![CleanShot 2022-04-28 at 16 42 29@2x](https://user-images.githubusercontent.com/12430020/165766005-2285ac0d-ae09-4cad-8e64-406e90868b8b.png)

**Comment by cpapazoglou** [comment]
Now we have launched Yoast Seo Premium, let's try tackling this. It is still depending on https://github.com/Automattic/wp-calypso/issues/63714.

cc @WBerredo

**Comment by gcsecsey** [comment]
We've discussed this ticket on our backlog grooming yesterday: p1659618475427209-slack-C02JPCHEKDY

> The paid plugins definition contains a `org_slug` which we can use to match a free with a paid plugin.
> 
> ![CleanShot 2022-02-18 at 12 41 29@2x](https://user-images.githubusercontent.com/12430020/154667439-2a001c8a-62dd-4fd4-854c-c8878a7eddc0.png)

@cpapazoglou should we consider reproducing the `org_slug` field on WC.com and sync that back through ES to be able to query the respective paid plugins? Or, based on the example of Yoast Premium, is the free --> paid plugin mapping will only happen with plugins that we create on WP.com?

**Comment by cpapazoglou** [comment]
WC.com should already include a `wpcom_marketplace_org_slug` meta product field 

https://github.com/Automattic/woocommerce.com/blob/2eecf7406c8adb139bd6ac92adb4ad56c8b2a443/plugins/wccom-plugins/wpcom-marketplace/includes/fields/class-wpcom-org-slug.php#L7

We will need to be able to use ES to get WC.com (premium) plugins with a given `wpcom_marketplace_org_slug`. Example:

- User visits Yoast SEO
- We search for premium related plugins in WC.com by using ES and `wpcom_marketplace_org_slug = wordpress_seo`
- Yoast SEO Premium is returned and shown / suggested to the user

**Comment by gcsecsey** [comment]
> WC.com should already include a `wpcom_marketplace_org_slug` meta product field
> 
> We will need to be able to use ES to get WC.com (premium) plugins with a given `wpcom_marketplace_org_slug`. Example:
> 
> * User visits Yoast SEO
> * We search for premium related plugins in WC.com by using ES and `wpcom_marketplace_org_slug = wordpress_seo`
> * Yoast SEO Premium is returned and shown / suggested to the user

Thanks for the detailed description @cpapazoglou. I've checked the field quickly from the site explorer (?type=postmeta&blog_id=113771570&post_id=390890), it doesn't seem to be synced yet. I've added the field to #63903 to tackle the sync of it together with the rest of the meta fields.

**Comment by gcsecsey** [comment]
After discussing this with @WBerredo we decided to use ES to solve this issue on the backend.

With a query like this, we can get the slug of the WC.com product that has the WP.org slug added to it:
```json
GET jetpack-search-25/_search
{
  "query": {
    "bool": {
      "must": [
        {
          "terms": {
            "blog_id": [
              "113771570"
            ]
          }
        },
        {
          "term": {
            "meta.wpcom_marketplace_org_slug.value": {
              "value": "mailpoet"
            }
          }
        }
      ]
    }
  },
  "_source": false,
  "fields": ["slug"]
}
```

Similarly to how we've added install numbers at indexing time to WC.com products in #68218, we can add the WC.com slug with the above query to a new field when indexing WP.org plugins.

**Comment by gcsecsey** [comment]
Diff created: D93134-code

I'll add a separate card to the board to track it separately from the Calypso changes.

**Comment by gcsecsey** [comment]
ES changes are deployed on this change: r256764-wpcom

I'll start a re-index for WP.org plugins to add the new fields to version 25 `jetpack-search`.

**Comment by gcsecsey** [comment]
Sorry, forgot to add that the indexing was done successfully, it took around 40 minutes. 😄

**Comment by gcsecsey** [comment]
Based on our discussion here: p1669761107188599-slack-C02JPCHEKDY

We still need to add the `premium_slug` field to the allowlist to be returned from the Marketplace search API.

Also, when checking the `jetpack-search-25` index, only some of the plugins have the `premium_slug` field added. Example query to check:
```json
GET jetpack-search-25/_search
{
  "query": {
    "bool": {
      "must": [
        {
          "terms": {
            "blog_id": ["108986944"]
          }
        },
        {
          "exists": {
            "field": "plugin.premium_slug"
          }
        }
      ]
    }
  },
  "_source": false,
  "fields": ["plugin.premium_slug"]
```

I'll raise a follow-up diff to update the allowlist and look into the indexing issue as well.

**Comment by gcsecsey** [comment]
Diff created for adding the `premium_slug` field to the API response: D93879-code

Also, I think this ongoing issue might be related to why we're not seeing more plugins with premium slugs assigned. Some plugins (eg. `woocommerce`, `wordpres-seo`) are currently missing from the `jetpack-search-25` index: p1669786925896139-slack-C029GN3KD

**Comment by epeicher** [comment]
Currently, the `premium_slug` is populated from the field `slug` that is populated from the `post_name` in this line: fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Scyhtvaf%2Srynfgvpfrnepu%2Svaqrkvat%2Sjcrf%2Qyvo%2Sfep%2Spbzzba%2Spynff.jcrf%2Qjc%2Qcbfg%2Qsvryq%2Qohvyqre.cuc%3Se%3Qs51s752r%26zb%3Q15808%26sv%3Q468%23468-og

That `post_name` comes from the `Permalink` in WooCommerce. For example for `Astra Pro` near the title we can see the permalink last value that will populate the `slug` returned by Elasticsearch. <img width="689" alt="CleanShot 2022-12-05 at 19 23 19@2x" src="https://user-images.githubusercontent.com/3519124/205713255-b349ca65-82b8-459e-9187-5664b5c1fb4d.png">

That means that the current implementation is correct. This is the line where the `slug` is used:
fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Scyhtvaf%2Srynfgvpfrnepu%2Svaqrkvat%2Sjcrf%2Qyvo%2Qjcpbz%2Spbzzba%2Spynff.jcrf%2Qjcpbz%2Qcyhtva%2Qcbfg%2Qsvryq%2Qohvyqre.cuc%3Se%3Qr3616np6%26zb%3Q2801%26sv%3Q99%23107-og

**Comment by epeicher** [comment]
Further to the previous comment, we just need to ensure that we use that `slug` to identify the product in the Marketplace catalog in both, the front-end and backend, respectively:
- When building the URL in Calypso [here](https://github.com/Automattic/wp-calypso/blob/b92f0fed6fc50f9e1396b31366428a75bfdf310c/client/my-sites/plugins/plugins-browser-item/index.jsx#L74)
- In the backend when finding the product from the catalog in the following line: fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Spynff%2Qznexrgcynpr%2Qcebqhpg%2Qpngnybt.cuc%3Se%3Qorqpr958%26zb%3Q11768%26sv%3Q362%23362-og

**Comment by epeicher** [comment]
Diff deployed r257447-wpcom

**Comment by epeicher** [comment]
The following products have been manually indexed to the `jetpack-search` index, and they now return the correct `plugin_slug`:
- [Visual Composer](https://wordpress.com/plugins/visualcomposer-wpcom)
- [Start Booking](https://wordpress.com/plugins/calendar-booking-wpcom)
- [Yoast SEO Premium](https://wordpress.com/plugins/wordpress-seo-premium)
- [Mailpoet Business](https://wordpress.com/plugins/mailpoet-business)

**Comment by WBerredo** [comment]
This task is currently dependent on #63171

**Comment by cpapazoglou** [comment]
Actually this depends on getting the .org plugins data from either the ES or the shadow site.

If we get from the ES, the `premium_slug` field is already populated while if we decide to get the data from the shadow site we will need to move the `premium_slug` implementation there.

Suggestion: create a function that uses ES in Calypso and returns a single .org plugin. Here is the function for getting all plugins https://github.com/Automattic/wp-calypso/blob/a03dacd98ce7c3bf4ad605a8cd31a10c8958574b/client/data/marketplace/use-es-query.ts#L147

For filtering by slug, we can check how it's done for `author` in https://github.com/Automattic/wp-calypso/blob/a03dacd98ce7c3bf4ad605a8cd31a10c8958574b/client/data/marketplace/search-api.ts#L117
